src/test/java/com/puppycrawl/tools/checkstyle/internal/testmodules/TestBeforeExecutionFileFilter.java

```diff
@@ -19,12 +19,19 @@
 
 package com.puppycrawl.tools.checkstyle.internal.testmodules;
 
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
 import com.puppycrawl.tools.checkstyle.api.BeforeExecutionFileFilter;
 
-public final class TestBeforeExecutionFileFilter implements BeforeExecutionFileFilter {
+public final class TestBeforeExecutionFileFilter extends AutomaticBean
```

Comment:
we can move all updates for ` extends AutomaticBean` to separate minor commit and merge it sooner.
all production classes that are from `implements BeforeExecutionFileFilter` , `implements AuditListener`, `implements Filter ` are beans, so making them more real is good at any time.